sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -69,7 +69,7 @@ namespace Azure { namespace Core {
     };
 
   public:
-#ifdef _MSC_VER
+#if defined(_MSC_VER)
```

Comment:
Should we also have compiler macros in `platforms.hpp`?

Reply:
What do you mean exactly?

Reply:
in this file https://github.com/Azure/azure-sdk-for-cpp/blob/c96440df2d97ac7efbc5a664d83b171e32b26692/sdk/core/azure-core/inc/azure/core/platform.hpp, we defined our own platform related macros such as `AZ_PLATFORM_WINDOWS` and `AZ_PLATFORM_POSIX`, rather than use those raw macros like `_WIN32`, `__unix__`.

I suggest we add some macros like `AZ_COMPILER_MSVC` instead of using `_MSC_VER` directly.

Reply:
I don't feel strong about it, so feel free to ignore.

Reply:
I wouldn't do it - `_MSC_VER` and other compilers are well defined as a single macro, while POSIX was not, so it makes sense to have `AZ_PLATFORM_POSIX`, but I would not make the macro called `AZ_PLATFORM_MSVC`. Especially that `_MSC_VER` has MSVC version, while our macrow would not.